core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1206,25 +1206,29 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
-  def lockWorkspace(workspaceName: WorkspaceName): Future[Int] = {
+  def lockWorkspace(workspaceName: WorkspaceName): Future[Boolean] = {
     //don't do the sam REST call inside the db transaction.
     getWorkspaceContext(workspaceName) flatMap { workspaceContext =>
       requireAccessIgnoreLockF(workspaceContext, SamWorkspaceActions.own) {
         //if we get here, we passed all the hoops
 
         dataSource.inTransaction { dataAccess =>
-          dataAccess.submissionQuery.list(workspaceContext).flatMap { submissions =>
-            if (!submissions.forall(_.status.isTerminated)) {
-              DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"There are running submissions in workspace $workspaceName, so it cannot be locked.")))
-            } else {
-              dataAccess.workspaceQuery.lock(workspaceContext.toWorkspaceName)
-            }
-          }
+          lockWorkspaceInternal(workspaceContext, dataAccess)
         }
       }
     }
   }
 
+  private def lockWorkspaceInternal(workspaceContext: Workspace, dataAccess: DataAccess) = {
+    dataAccess.submissionQuery.list(workspaceContext).flatMap { submissions =>
+      if (!submissions.forall(_.status.isTerminated)) {
+        DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"There are running submissions in workspace ${workspaceContext.toWorkspaceName}, so it cannot be locked.")))
+      } else {
+        dataAccess.workspaceQuery.lock(workspaceContext.toWorkspaceName)
+      }
+    }
+  }
```

Comment:
You can do this in a for-comp which I think makes it easier to read? 
```suggestion
  private def lockWorkspaceInternal(workspaceContext: Workspace, dataAccess: DataAccess): ReadWriteAction[Boolean] =
    for {
      submissions <- dataAccess.submissionQuery.list(workspaceContext)
      _ <- Applicative[ReadWriteAction].unlessA(submissions.forall(_.status.isTerminated)) {
        throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, 
          s"Cannot lock workspace '${workspaceContext.toWorkspaceName}' as there are submissions running."
        ))
      }
      locked <- dataAccess.workspaceQuery.lock(workspaceContext.toWorkspaceName)
    } yield locked
```

Reply:
I find the applicative stuff less approachable than a common if statement.
Also this is preexisting code, just moved. I am not going to rewrite it.

Reply:
Give yourself to the applicative side of the force!!!